packages/flutter_test/lib/src/binding.dart

```diff
@@ -494,9 +494,25 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
   ///
   /// When [handlePointerEvent] is called directly, [pointerEventSource]
   /// is [TestBindingEventSource.device].
+  ///
+  /// This default means that pointer events triggered by the [WidgetController]
+  /// will result in actual interactions with the UI, but other pointer events
+  /// such as those from physical taps will be delegated to
+  /// [deviceEventDispatcher] *instead*. See also
+  /// [shouldPropagateDevicePointerEvents] if this is undesired.
   TestBindingEventSource get pointerEventSource => _pointerEventSource;
   TestBindingEventSource _pointerEventSource = TestBindingEventSource.device;
 
+  /// Whether pointer events from [TestBindingEventSource.device] will be
+  /// propagated to the framework, or dropped.
+  ///
+  /// Setting this can be useful to interact with the app in some other way
+  /// besides through the [WidgetController], such as with `adb shell input tap`
+  /// on Android.
+  ///
+  /// See also [pointerEventSource].
+  bool shouldPropagateDevicePointerEvents = false;
```

Comment:
nit: maybe drop the `should` from the name?

Reply:
[Effective Dart](https://dart.dev/guides/language/effective-dart/design#prefer-a-non-imperative-verb-phrase-for-a-boolean-property-or-variable) tells us to use prefix it with a `should` though. I don't feel very strongly about this though, happy to remove it.